SDK fullnode stop due to tomox snapshot failed

```golang 
DEBUG[08-22|04:33:55] Snapshot process takes                   time=36.78µs   hash=1d9635…feff04
panic: interface conversion: interface {} is *types.Trade, not *tomox.Item

goroutine 508 [running]:
github.com/ethereum/go-ethereum/tomox.(*Tree).GetNode(0xc008c4ef00, 0xc00361e280, 0x20, 0x20, 0xc01259ad00, 0x2362420, 0xc00b4439d0, 0xb6310e)
        /home/ubuntu/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/tomox/redblacktree.go:134 +0x1c9
github.com/ethereum/go-ethereum/tomox.(*Tree).Root(0xc008c4ef00, 0x1037b00, 0x14a5590)
        /home/ubuntu/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/tomox/redblacktree.go:43 +0x67
github.com/ethereum/go-ethereum/tomox.(*OrderTree).Save(0xc00b8a0e60, 0x459c00, 0x0, 0x0)
        /home/ubuntu/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/tomox/ordertree.go:75 +0x52
github.com/ethereum/go-ethereum/tomox.(*OrderBook).Restore(0xc00b8a0dc0, 0x0, 0x14f1d40, 0xc00027c040)
        /home/ubuntu/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/tomox/orderbook.go:133 +0x168
github.com/ethereum/go-ethereum/tomox.(*TomoX).getAndCreateIfNotExisted(0xc0000df7a0, 0xc012878300, 0x8, 0x0, 0x7fda5abe9648, 0x0, 0x4625a8)
        /home/ubuntu/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/tomox/tomox.go:669 +0xe4
github.com/ethereum/go-ethereum/tomox.(*TomoX).GetOrderBook(...)

```

--- a/core/blockchain.go
+++ b/core/blockchain.go
@@ -1235,7 +1235,7 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 						}
 					}
 				}
-				if bc.CurrentHeader().Number.Uint64()%common.TomoXSnapshotInterval == 0 {
+				if bc.CurrentHeader().Number.Uint64()%common.TomoXSnapshotInterval == 0 && !tomoXService.IsSDKNode() {
 					if err := tomoXService.Snapshot(block.Hash()); err != nil {
 						log.Error("Failed to snapshot tomox", "err", err)
 					}
@@ -1463,7 +1463,7 @@ func (bc *BlockChain) insertBlock(block *types.Block) ([]interface{}, []*types.L
 					}
 				}
 			}
-			if bc.CurrentHeader().Number.Uint64()%common.TomoXSnapshotInterval == 0 {
+			if bc.CurrentHeader().Number.Uint64()%common.TomoXSnapshotInterval == 0 && !tomoXService.IsSDKNode() {
 				if err := tomoXService.Snapshot(block.Hash()); err != nil {
 					log.Error("Failed to snapshot tomox", "err", err)
 				}
@@ -1960,7 +1960,6 @@ func logDataToSdkNode(tomoXService *tomox.TomoX, transactions types.Transactions
 	sort.Slice(txs, func(i, j int) bool {
 		return txMatchBatchData[txs[i].Hash()].Timestamp <= txMatchBatchData[txs[j].Hash()].Timestamp
 	})
-
 
 	for _, tx := range txs {
 		txMatchBatch, ok := txMatchBatchData[tx.Hash()]

--- a/tomox/tomox.go
+++ b/tomox/tomox.go
@@ -169,8 +169,10 @@ func New(cfg *Config) *TomoX {
 			}
 		},
 	}
-	if err := tomoX.loadSnapshot(common.Hash{}); err != nil {
-		log.Error("Failed to load tomox snapshot", "err", err)
+	if !tomoX.sdkNode {
+		if err := tomoX.loadSnapshot(common.Hash{}); err != nil {
+			log.Error("Failed to load tomox snapshot", "err", err)
+		}
 	}
 
 	return tomoX
@@ -1148,7 +1150,6 @@ func (tomox *TomoX) Snapshot(blockHash common.Hash) error {
 	var (
 		snap *Snapshot
 		err  error
-		blob interface{}
 	)
 	defer func(start time.Time) {
 		if err != nil {
@@ -1164,21 +1165,11 @@ func (tomox *TomoX) Snapshot(blockHash common.Hash) error {
 	if err = snap.store(tomox.db); err != nil {
 		return err
 	}
-	// get current snapshot hash in database
-	oldHash := common.Hash{}
-	if blob, err = tomox.db.Get([]byte(latestSnapshotKey), &common.Hash{}, false); err == nil && blob != nil {
-		oldHash = *blob.(*common.Hash)
-	}
+
 	if err = tomox.db.Put([]byte(latestSnapshotKey), &blockHash, false); err != nil {
 		return err
 	}
 
-	// remove old snapshot
-	if oldHash != (common.Hash{}) {
-		if err = tomox.db.Delete(append([]byte(snapshotPrefix), oldHash[:]...), false); err != nil {
-			return err
-		}
-	}
 	return nil
 }
 

